### PR TITLE
Upgrade testkit to 0.10, update Github workflows

### PR DESCRIPTION
**Pull request by srtfisher**

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -16,16 +16,15 @@
...
+    "mantle-framework/testkit": "^0.10"
   },
   "config": {
     "allow-plugins": {
       "dealerdirect/phpcodesniffer-composer-installer": true,
       "alleyinteractive/composer-wordpress-autoloader": true
     },
     "platform": {
-      "php": "7.4"
+      "php": "8.0.26"
```
Comment (kevinfodness) [inline composer.json:27]: Why the specific version number? Could we use 8.0 instead?

**Review by kevinfodness — APPROVED** [review]
See my comment about the specific PHP version. Also, the `.bak` file for the old PHPUnit config should be removed before merge. Otherwise 🍣

File: composer.json (hunk 1 above)
Comment (srtfisher) [reply composer.json:27]: A number of packages actually require > 8.0.4

Comment (kevinfodness) [reply composer.json:27]: (for my own education) what would happen if this was installed on a platform running 8.0.27? Is this considered a minimum required version?

**Pull request merged**